ironfish/src/rpc/routes/chain/getNoteWitness.ts

```diff
@@ -46,10 +50,22 @@ router.register<typeof GetNoteWitnessRequestSchema, GetNoteWitnessResponse>(
   GetNoteWitnessRequestSchema,
   async (request, node): Promise<void> => {
     const { chain } = node
-    const witness = await chain.notes.witness(request.data.index)
+
+    const confirmations = request.data.confirmations ?? node.config.get('confirmations')
+
+    const maxConfirmedSequence = Math.max(
+      chain.head.sequence - confirmations,
+      GENESIS_BLOCK_SEQUENCE,
+    )
+    const maxConfirmedHeader = await chain.getHeaderAtSequence(maxConfirmedSequence)
+
+    Assert.isNotNull(maxConfirmedHeader)
+    Assert.isNotNull(maxConfirmedHeader?.noteSize)
+
+    const witness = await chain.notes.witness(request.data.index, maxConfirmedHeader.noteSize)
 
     if (witness === null) {
-      throw new ValidationError(`No notes exist with index ${request.data.index}`)
+      throw new ValidationError(`No confirmed notes exist with index ${request.data.index}`)
```

Comment:
I wonder if it'd be worth logging the tree size here as well? So if a note is past the confirmation range, it would somehow indicate the reason why witness creation failed

Reply:
I think that's a good idea; better to include more information if it's available!